platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/RuntimeTypeAdapterFactory.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2011 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Source: https://github.com/google/gson/blob/11b273241250c3103a6ec52b45f792328f50479c/extras/src/main/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactory.java
```

Comment:
❌ I think there should be a comment, why we need this and why it's added this way

Reply:
Makes sense, I'll add explanation if I find that we _really_ need this class.